episodes/multi-dimensional-images.md

```diff
@@ -0,0 +1,449 @@
+
+![](fig/brain-napari.png){alt="A screenshot of a head X-ray in Napari"}
+
+This image shows part of a human head X-ray CT dataset. We can see it has three 
+dimensions due to the slider at the base of the image, and the shape output:
+
+```python
+image = viewer.layers["brain"].data
+print(image.shape)
+```
+
+```output
+# (z, y, x)
+(10, 256, 256)
+```
+
+This 3D image can be thought of as a stack of 2D images, where the z axis points 
+along the stack. It is stored as a 3D array:
+
+![](fig/2d-3d-arrays.png){alt="A diagram comparing 2D and 
+3D image arrays" width='80%'}
+
+In Napari (and Python in general), dimensions are referred to by their index 
+e.g. here dimension 0 is the z axis, dimension 1 is the y axis and dimension 2 
+is the x axis. We can check this in Napari by looking at the number at the very 
+left of the slider. Here it's labelled '0', showing that it controls movement 
+along dimension 0 (i.e. the z axis).
+
+## Channels
+
+Next, let's look at a 3D image with channels. Remove the brain image and select:  
+`File > Open Sample > napari builtins > Cells (3D+2Ch)`  
+
+![](fig/cells-napari.png){alt="A screenshot of a flourescence microscopy image 
+of some cells in Napari"}
+
+Channels are used e.g. with fluorescence microscopy to store images collected 
+with different wavelengths of light. In the currently open image, we can see 
+two channels - one for a fluorescent label targeting nuclei, and another for a 
+fluorescent label targeting cell membranes. These channels are shown as separate 
+image layers in Napari's layer list.
+
+This image has a total of 4 dimensions (czyx), so why is there only one slider? 
+This is because Napari automatically recognises the channels and separates them 
+into different image layers. We can see what happens when Napari doesn't 
+recognise the channels, by selecting both image layers (shift + click so they're 
+both highlighted in blue), right clicking on them and selecting:  
+`Merge to stack`
+
+You should see the two image layers disappear, and a new combined layer appear 
+in the layer list (labelled 'membrane'). This image has an extra slider that 
+allows switching channels - try moving both sliders and see how the image 
+display changes.
+
+We can check the image's dimensions with:
+
+```python
+
+image = viewer.layers["membrane"].data
+print(image.shape)
+```
+
+```output
+# (c, z, y, x)
+(2, 60, 256, 256)
+```
+
+This clearly shows all 4 dimensions, with dimension 0 as channels and dimension 
+1 as z. This matches the labels at the left hand side of each of the two 
+sliders. We can separate the channels again by right clicking on the 'membrane' 
+image layer and selecting:  
+`Split Stack`
+
+Given its 4 dimensions, this image is stored in a 4D image array. This is 
+similar to having one 3D image array per channel, stacked side by side:
+
+![](fig/3d-4d-arrays.png){alt="A diagram comparing 3D and 4D 
+image arrays" width='80%'}
+
```

Comment:
I'm getting a little confused about layers vs channels? You could have a 3 channel image (rgb) with a single layer. I need to go over this again and make sure the language is consistent. I'm not entirely sure myself what the difference is between channels and layers.